src/index.js

```diff
@@ -1,21 +1,28 @@
 import React, { useState } from 'react';
 import ReactDOM from 'react-dom';
-import './Styles/App.scss';
+import './styles/App.scss';
 import Header from './Header/Header';
 import Introduction from './Introduction/Introduction';
 import CardsList from './CardsList/CardsList';
 import Search from './Search/Search';
 import CardList from './CardsInfo.json';
+import CardsContext from './Contexts/CardsContext';
+import HeaderContext from './Contexts/HeaderContext';
 
 const App = () => {
-  const [items, setItems] = useState(CardList.cards);
+  const [cards, setCards] = useState(CardList.cards);
+  const [showMobileNav, setShowMobileNav] = useState(false);
 
   return (
     <>
-      <Header />
+      <HeaderContext.Provider value={[showMobileNav, setShowMobileNav]}>
+        <Header />
+      </HeaderContext.Provider>
       <Introduction />
-      <Search setItems={setItems} items={items} />
-      <CardsList items={items} />
+      <CardsContext.Provider value={[cards, setCards]}>
+        <Search />
+        <CardsList />
+      </CardsContext.Provider>
```

Comment:
That's definitely a way to do it. But not the optimal one. You can place search under cards to have scoped state. You don't need **React** state to manage CSS.